File: 2025/July-20-2025/LongestSub.js
// Given a string s, find the length of the longest substring without duplicate characters.

// Example 1:

// Input: s = "abcabcbb"
// Output: 3
// Explanation: The answer is "abc", with the length of 3.

var s = "abcabcbb"

const longestSub = (s) => { 
  let left = 0; 
  let seen = new Set()
  let maxLength = 0; 

  for(let right = 0; right < s.length; right++){
   while(seen.has(s[right])) { 
    seen.delete(s[left])
    left++
   }
   seen.add(s[right])
   maxLength = Math.max(maxLength, right - left + 1)
  }
  return maxLength
}

console.log(longestSub(s))

File: 2026/January/LongestSub.py
# Longest Substring Without Repeating Characters
# Solved 
# Given a string s, find the length of the longest substring without duplicate characters.

# A substring is a contiguous sequence of characters within a string.

# Example 1:

# Input: s = "zxyzxyz"

# Output: 3


s = "zxyzxyz"

def longest_sub(s: str) -> int: 
  left = 0
  check = set()
  max_len = 0

  for right in range(len(s)): 
    if s[right] in check: 
      check.remove(s[left])
      left += 1
    check.add(s[right])
    max_len = max(max_len, right - left + 1)
  return max_len

print(longest_sub(s))
   


  